### PR TITLE
docs(components): [cascader]

### PR DESCRIPTION
**Pull request by chenxch**
* Update cascader docs with new syntax.

image

image

image

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/en-US/component/cascader.md [hunk 1]
```diff
@@ -142,96 +142,103 @@ cascader/panel
...
+| before-filter                          | hook function before filtering with the value to be filtered as its parameter. If `false` is returned or a `Promise` is returned and then is rejected, filtering will be aborted | ^[Function]`(value: string) => boolean`                       | —       |
+| popper-class                           | custom class name for Cascader's dropdown                                                                                                                                        | ^[string]                                                     | ''      |
+| teleported                             | whether cascader popup is teleported                                                                                                                                             | ^[boolean]                                                    | true    |
+| popper-append-to-body<DeprecatedTag /> | whether to append the popper menu to body. If the positioning of the popper is wrong, you can try to set this prop to false                                                      | ^[boolean]                                                    | true    |
```
Comment (jw-foss) [inline docs/en-US/component/cascader.md:168]: We should update this to our new syntax, refer to #11580

File: docs/en-US/component/cascader.md [hunk 2]
```diff
@@ -142,96 +142,103 @@ cascader/panel
...
+| expandTrigger  | trigger mode of expanding options                                                                          | ^[enum]`'click' \| 'hover'`                         | click    |
+| multiple       | whether multiple selection is enabled                                                                      | ^[boolean]                                          | false    |
+| checkStrictly  | whether checked state of a node not affects its parent and child nodes                                     | ^[boolean]                                          | false    |
+| emitPath       | when checked nodes change, whether to emit an array of node's path, if false, only emit the value of node. | ^[boolean]                                          | true     |
+| lazy           | whether to dynamic load child nodes, use with `lazyload` attribute                                         | ^[boolean]                                          | false    |
+| lazyLoad       | method for loading child nodes data, only works when `lazy` is true                                        | ^[Function]`(node: Node, resolve: Resolve) => void` | —        |
```
Comment (jw-foss) [inline docs/en-US/component/cascader.md:238]: **Suggestion:** maybe we should specify this `Resolve` type, this could be a little bit confusing.

**Comment by jw-foss** [comment]
@chenxch what do you think about adding a section for specifying the types like: https://vueuse.org/core/useAsyncState/#type-declarations did, since some types were not that clear, and if we want to know the exact type we will be forced to read the source code and that is something we really need to avoid.

File: docs/en-US/component/cascader.md (hunk 1 above)
Comment (chenxch) [reply docs/en-US/component/cascader.md]: done.

**Comment by chenxch** [comment]
@jw-foss Awesome, we need this, I'm going to add it to our document now.

**Comment by chenxch** [comment]
image
@jw-foss

**Comment by chenxch** [comment]
> 

image
@jw-foss Which do you think we should use?

**Review by jw-foss — APPROVED** [review]

**Pull request merged**